addons/stock/models/stock_picking.py

```diff
@@ -1000,11 +1007,11 @@ def button_validate(self):
             return res
 
         # Call `_action_done`.
+        pickings_not_to_backorder = self.filtered(lambda p: p.picking_type_id.create_backorder == 'never')
         if self.env.context.get('picking_ids_not_to_backorder'):
-            pickings_not_to_backorder = self.browse(self.env.context['picking_ids_not_to_backorder'])
+            pickings_not_to_backorder += self.browse(self.env.context['picking_ids_not_to_backorder'])
```

Comment:
```suggestion
            pickings_not_to_backorder |= self.browse(self.env.context['picking_ids_not_to_backorder'])
```
To avoid to have the same record multiple time in the same recordset.